here-naksha-lib-base/src/jvmMain/kotlin/naksha/base/Platform.kt

```diff
@@ -512,11 +512,21 @@ actual class Platform {
                 if (klass.isInstance(proxy)) return proxy as T
                 if (doNotOverride) throw IllegalStateException("The symbol $symbol is already bound to incompatible type")
             }
-            proxy = klass.primaryConstructor!!.call()
+
+            proxy = nonArgConstructorFor(klass).call()
```

Comment:
Please add the caching, should be just a few code lines, but avoids to always go through reflection and search through constructor lists.